src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java

```diff
@@ -170,31 +160,20 @@ private String extractRequestBody(StaplerRequest req) {
         return body;
     }
 
-    
-    private Set<GhprbWebHook> getWebHooks() {
-        final Set<GhprbWebHook> webHooks = new HashSet<GhprbWebHook>();
-
-        // We need this to get access to list of repositories
-        Authentication old = SecurityContextHolder.getContext().getAuthentication();
-        SecurityContextHolder.getContext().setAuthentication(ACL.SYSTEM);
-
-        try {
-            for (AbstractProject<?, ?> job : Jenkins.getInstance().getAllItems(AbstractProject.class)) {
-                GhprbTrigger trigger = job.getTrigger(GhprbTrigger.class);
-                if (trigger == null || trigger.getWebHook() == null) {
-                    continue;
+    private Set<GhprbTrigger> getTriggers(String repoName, String body, String signature) {
+        Set<GhprbTrigger> triggers = new HashSet<GhprbTrigger>();
+        
+        Set<AbstractProject<?, ?>> projects = GhprbTrigger.getDscp().getRepoTriggers(repoName);
+        if (projects != null) {
+            for (AbstractProject<?, ?> project : projects) {
+                GhprbTrigger trigger = Ghprb.extractTrigger(project);
+                if (trigger.matchSignature(body, signature)) {
+                    triggers.add(trigger);
                 }
-                webHooks.add(trigger.getWebHook());
             }
-        } finally {
-            SecurityContextHolder.getContext().setAuthentication(old);
```

Comment:
Honestly I don't know.  I was assuming it was needed to iterate through all of the projects in Jenkins.   Do you know how this could be tested?